core/o11y/src/macros.rs

```diff
@@ -6,38 +6,42 @@ macro_rules! handler_span_use_context {
 /// Use this at the beginning of `handle` methods to reduce the boilerplate.
 #[macro_export]
 macro_rules! handler_span {
-    ($target:expr, $actor:expr, $handler:expr, $msg:expr) => {{
-        let reflection_actor =
-            std::any::type_name::<Self>().rsplit_once("::").map_or("__", |(_, name)| name);
-        let span = tracing::debug_span!(
-            target: $target,
-            "handle",
-            handler = $handler,
-            actor = $actor,
-            reflection_actor = reflection_actor
-        )
-        .entered();
-
+    ($target:expr, $msg:expr) => {{
         let WithSpanContext { msg, context, .. } = $msg;
+
+        fn fix_name(name: &str) -> &str {
+            name.rsplit_once("::").map_or("_", |(_, name)| name)
+        }
+        fn type_name_of<T>(_: &T) -> &str {
+            fix_name(std::any::type_name::<T>())
+        }
```

Comment:
Rather than defining these functions in every macro expansion, define them outside of the macro and use `$crate::fix_name(...)` to refer to the helper within macro. Otherwise we’re going to be generating two full functions for every macro invocation, which is probably not going to be very good compile-time wise.